src/databases/indexedDb/loadRecords.ts

```diff
@@ -121,16 +121,67 @@ export async function loadRecords(countUpdater: ((totalCount: number) => void) |
 
 	if (countUpdater) void updateCount(collectionForCount, countUpdater);
 
-	if (needsMemorySort && sort) {
-		let items: object[];
-
-		if (sort.desc) processedCollection = processedCollection.reverse();
-		items = await processedCollection.sortBy(sort.id);
+	// Determine if there are out-of-line keys
+	const primKeySchema = table.schema.primKey;
+	const keyPath = primKeySchema.keyPath;
+	const isOutboundKey = (keyPath as string | string[] | null) === null;
```

Comment:
I think this condition got accidentally modified when you fix conflict

```suggestion
	const primKeyPath = table.schema.primKey.keyPath;
	const primaryKey = Array.isArray(primKeyPath)
		? (primKeyPath[0])
		: (primKeyPath);
	const hasOutOfLineKeys = !primaryKey?.trim();
```